src/utils/queryBuilder.ts

```diff
@@ -113,104 +115,121 @@ const buildGeoShapeQuery = (geoCoordinates: IGeoCoordinates): IGeoShapeQuery =>
   return geoShapeQuery;
 };
 
-const buildSortQuery = (finalQuery: IQuery, sort: string, isCount: boolean): IQuery => {
-  if (sort && !isCount) {
-    const sortQuery: ISortQuery =
-      sort === 'recent_study' ? buildCustomSortScriptForStudyPeriod() : buildBestScoreSort();
-    finalQuery.sort?.push(sortQuery);
-  } else {
-    delete finalQuery.sort;
-  }
-  return finalQuery;
-};
-
-const buildAggregationQuery = (
-  finalQuery: IQuery,
-  ignoreAggregation: boolean,
-  isCount: boolean,
-  aggregationField: string,
-): IQuery => {
-  if (!ignoreAggregation && !isCount && aggregationField) {
-    const aggregateQuery: IAggregateQuery = {
-      unique_values: {
-        terms: {
-          field: aggregationField,
-        },
-      },
-    };
-    finalQuery.aggs = aggregateQuery;
-    finalQuery.size = 0;
-  } else {
-    delete finalQuery.aggs;
-  }
-
-  return finalQuery;
-};
-
-const buildSearchQuery = (
-  searchFieldsObject: ISearchPayload,
-  fieldsToSearch: string[] = [],
-  isCount: boolean = false,
-  ignoreAggregation: boolean = false,
-  aggregationField: string = '',
-): IQuery => {
-  const { fields, sort, rowsPerPage, page, filters: filterOptions } = searchFieldsObject;
+const buildSearchQuery = (searchBuilderPayload: ISearchBuilderPayload): IQuery => {
+  const {
+    searchFieldsObject,
+    fieldsToSearch = [],
+    isCount = false,
+    ignoreAggregation = false,
+    aggregationField = '',
+    docId = '',
+  } = searchBuilderPayload;
   const boolQuery: IBoolQuery = {
     bool: {
       must: [],
     },
   };
+  const { fields, sort, rowsPerPage, page, filters: filterOptions } = (searchFieldsObject as ISearchPayload) ?? {};
+
+  addKeywordSearchQuery(fields, fieldsToSearch, boolQuery);
+  addFilterOptionsQuery(filterOptions, isCount, ignoreAggregation, boolQuery);
+  addDateSearchQuery(fields, boolQuery);
+  addCoordinateSearchQuery(fields, boolQuery);
+  !Object.keys(searchFieldsObject ?? {}).length && docId && addDetailsQuery(docId, boolQuery);
+
+  const isSort = sort && !isCount;
+  const isAggregation = !ignoreAggregation && !isCount && aggregationField;
+
+  const finalQuery: IQuery = {
+    query: boolQuery,
+    ...(isSort && { sort: [] }),
+    ...(isAggregation && { aggs: {} }),
+    ...(!isCount && { size: isAggregation ? 0 : rowsPerPage }),
```

Comment:
rowsPerPage is number as per the interface but passed as string from the customScripts.js
rows per page works as expected?

Reply:
Applied as mentioned [above](https://github.com/DEFRA/ncea-frontend/pull/25#discussion_r1519561589).

Reply:
I am not convinced. we are multiplying the rowPerPage, not sure typescript automatically does the type casting

Reply:
While invoking the ajax call, the rowsPerPage property is being sent as an integer. We are doing that here [here ](https://github.com/DEFRA/ncea-frontend/blob/5bfbd1ed48852d3d73d97fe1b6916958b39dd15d/public/scripts/fetchResults.js#L198)